plugins/modules/system/java_keystore.py

```diff
@@ -150,31 +148,36 @@ def read_stored_certificate_fingerprint(module, keytool_bin, alias, keystore_pat
     (rc, stored_certificate_fingerprint_out, stored_certificate_fingerprint_err) = run_commands(
         module, stored_certificate_fingerprint_cmd, environ_update=dict(STOREPASS=keystore_password))
     if rc != 0:
-        if "keytool error: java.lang.Exception: Alias <%s> does not exist" % alias not in stored_certificate_fingerprint_out:
-            return module.fail_json(msg=stored_certificate_fingerprint_out,
-                                    err=stored_certificate_fingerprint_err,
-                                    rc=rc,
-                                    cmd=stored_certificate_fingerprint_cmd)
-        else:
-            return None
```

Comment:
This `else` branch no longer exists in the code, which causes some changed behavior.

Reply:
No, in fact it doesn't. This `else` branch is inside a parent `if rc !=0:` block that is currently never reached due to `check_rc=True`.

Keeping this `else` branch design would mean: if the named alias does not exist, overwrite the keystore, that is not the current/legacy behavior (module fails, despite the fact that a unit test validates the opposite behavior). The PR doesn't change this: the module still fails when the named alias does not exist.

Reply:
I totally forgot about the `check_rc` change... Thanks!